Small documentation update tweaks (#1782)
mono/linker: docs/data-formats.md

# Data Formats

## Input Data Formats

> ILLinker uses several data formats to control or influnce the linking process. The data formats are not versioned but are backward compatible.

Reviewer: ```suggestion
ILLink uses several data formats to control or influence the linking process. The data formats are not versioned but are backward compatible.
```

Notice that a descriptor file containing the custom attribute type overrides this behavior. In case the
custom attribute type is being referenced in a descriptor file and in the attribute annotations file
for removal, the custom attribute will not be removed.

## Dependencies Trace Format

This is the format of data used to capture linker logic about why

> members, type and other metadata elements were marked by the linker

Reviewer: ```suggestion
members, types, and other metadata elements were marked by the linker
```

> as required and persisted in the linked output. The format output edges

Reviewer: ```suggestion
as required and persisted in the linked output. The format includes edges
```
